### PR TITLE
fix(api): Organization Slug Trailing Space

### PR DESCRIPTION
**Pull request by mgaeta**
Add migration to trim trailing space. See [Sentry issue](https://sentry.io/share/issue/06f28151cc6844aaa2ce4b0cb8330f0d).

### CODE REVIEW

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
Blocking this now because we can’t run this in production. Will defer to @wedamija for more.

File: src/sentry/migrations/0107_remove_spaces_from_slugs.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    atomic = True
```
Comment (wedamija) [inline src/sentry/migrations/0107_remove_spaces_from_slugs.py:23]: When we change this, set `atomic = False`

**Review by wedamija — COMMENTED** [review]
Since this table is relatively small at ~400k rows, and there are only two rows affected total, I think we can filter in the database and then update each row individually. So just make this into a Django query that uses the query you have here, and then update each model instance individually. Just a little safer than running the update on the whole table, not sure what the lock situation is going to be like if we use this statement.

File: src/sentry/migrations/0107_remove_spaces_from_slugs.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    dependencies = [("sentry", "0106_service_hook_project_id_nullable")]
+
+    operations = [
+        migrations.RunSQL(
+            "UPDATE TABLE sentry_organization SET name=TRIM(slug) WHERE slug like '% '"
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/migrations/0107_remove_spaces_from_slugs.py:29]: Just saw @wedamija's review. Given there are only two rows to change RangeQuerySet is going to be overkill. Fetching those rows and doing individual updates will work.

**Comment by mattrobenolt** [comment]
Yeah, it's more that an UPDATE in a single query over the entire organization table is going to lock the whole table. There's no index on what you're filtering on, so it's scanning every row.

Even if it's slower, it's better and safer here to just use the ranged wrapper to iterate every row and just check in Python.

File: src/sentry/migrations/0107_remove_spaces_from_slugs.py (hunk 1 above)
Comment (mgaeta) [reply src/sentry/migrations/0107_remove_spaces_from_slugs.py]: 👍

File: src/sentry/migrations/0107_remove_spaces_from_slugs.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
```
[the remark this answers is not in the document]
Comment (mgaeta) [reply src/sentry/migrations/0107_remove_spaces_from_slugs.py]: 👍

**Comment by mgaeta** [comment]
Rebasing to fix conflicts and Travis.

**Review by markstory — APPROVED** [review]

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**